File: backend/controllers/systemController.js
const axios = require('axios');

const DOTNET_SERVICE_URL = 'http://localhost:5192/api/system';

// @desc    Trigger Reconciliation
// @route   POST /api/system/reconcile
// @access  Private/Admin
const reconcile = async (req, res) => {
    try {
        const response = await axios.post(`${DOTNET_SERVICE_URL}/reconcile`);
        res.json(response.data);
    } catch (error) {
        console.error(error);
        res.status(500).json({ message: 'Error communicating with .NET service' });
    }
};

// @desc    Trigger Productivity Calculation
// @route   POST /api/system/calculate-productivity
// @access  Private/Admin
const calculateProductivity = async (req, res) => {
    try {
        const response = await axios.post(`${DOTNET_SERVICE_URL}/calculate-productivity`);
        res.json(response.data);
    } catch (error) {
        console.error(error);
        res.status(500).json({ message: 'Error communicating with .NET service' });
    }
};

// @desc    Trigger Rule Evaluation
// @route   POST /api/system/evaluate-rules
// @access  Private/Admin
const evaluateRules = async (req, res) => {
    try {
        const response = await axios.post(`${DOTNET_SERVICE_URL}/evaluate-rules`);
        res.json(response.data);
    } catch (error) {
        console.error(error);
        res.status(500).json({ message: 'Error communicating with .NET service' });
    }
};

// @desc    Run All System Checks
// @route   POST /api/system/run-all
// @access  Private/Admin
const runAll = async (req, res) => {
    try {
        const response = await axios.post(`${DOTNET_SERVICE_URL}/run-all`);
        res.json(response.data);
    } catch (error) {
        console.error(error);
        res.status(500).json({ message: 'Error communicating with .NET service' });
    }
};

module.exports = { reconcile, calculateProductivity, evaluateRules, runAll };

File: dotnet_service/Controllers/SystemController.cs
using Microsoft.AspNetCore.Mvc;
using System.Threading.Tasks;
using UnifiedEmployeeSystem.Service.Services;

namespace UnifiedEmployeeSystem.Service.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class SystemController : ControllerBase
    {
        private readonly ReconciliationService _reconciliationService;
        private readonly ProductivityService _productivityService;
        private readonly RuleEngineService _ruleEngineService;

        public SystemController(
            ReconciliationService reconciliationService,
            ProductivityService productivityService,
            RuleEngineService ruleEngineService)
        {
            _reconciliationService = reconciliationService;
            _productivityService = productivityService;
            _ruleEngineService = ruleEngineService;
        }

        [HttpPost("reconcile")]
        public async Task<IActionResult> Reconcile()
        {
            await _reconciliationService.ReconcileAsync();
            return Ok(new { message = "Reconciliation completed" });
        }

        [HttpPost("calculate-productivity")]
        public async Task<IActionResult> CalculateProductivity()
        {
            await _productivityService.CalculateProductivityAsync();
            return Ok(new { message = "Productivity calculation completed" });
        }

        [HttpPost("evaluate-rules")]
        public async Task<IActionResult> EvaluateRules()
        {
            await _ruleEngineService.EvaluateRulesAsync();
            return Ok(new { message = "Rule evaluation completed" });
        }
        
        [HttpPost("run-all")]
        public async Task<IActionResult> RunAll()
        {
             await _reconciliationService.ReconcileAsync();
             await _productivityService.CalculateProductivityAsync();
             await _ruleEngineService.EvaluateRulesAsync();
             return Ok(new { message = "All system checks completed" });
        }
    }
}
